src/main/java/org/opensearch/knn/index/MethodComponentContext.java

```diff
@@ -193,6 +200,92 @@ public Map<String, Object> getParameters() {
         return parameters;
     }
 
+    /**
+     *
+     * Provides a String representation of MethodComponentContext
+     * Sample return:
+     * {name=ivf;parameters=[nlist=4;type=fp16;encoder={name=sq;parameters=[nprobes=2;clip=false;]};]}
+     *
+     * @return string representation
+     */
+    @Override
+    public String toString() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("{name=").append(name).append(DELIMITER);
+        stringBuilder.append("parameters=[");
+        if (Objects.nonNull(parameters)) {
+            for (Map.Entry<String, Object> entry : parameters.entrySet()) {
+                stringBuilder.append(entry.getKey()).append("=");
+                String value = entry.getValue().toString();
+                // Model Metadata uses a delimiter to split the input string in its fromString method
```

Comment:
Link to ModelMetadata code and line number for reference (2.12 tag). This will allow future devs to see exactly where this is happening